sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -358,7 +353,8 @@ object JavaTypeInference {
    */
   def serializerFor(beanClass: Class[_]): Expression = {
     val inputObject = BoundReference(0, ObjectType(beanClass), nullable = true)
-    val nullSafeInput = AssertNotNull(inputObject, Seq("top level input bean"))
+    val nullSafeInput = AssertNotNull(inputObject,
+      WalkedTypePath().recordRoot("top level input bean").getPaths)
```

Comment:
nit: we can keep it unchanged.

Reply:
Ah yes that's not even same. Will revert.